opennlp-tools/src/main/java/opennlp/tools/parser/Parse.java

```diff
@@ -518,6 +518,7 @@ public void add(Parse daughter, HeadRules rules) {
     this.head = rules.getHead(getChildren(),type);
     if (head == null) {
       System.err.println(parts);
+      throw new IllegalArgumentException("Rule returned null head!");
```

Comment:
Yup, I wasn't sure how to fix this one... if `head` is `null`, ATM, we get a `NullPointerException` and that's it.

I threw this exception as somewhere near I saw this same exception being used... but it indeed needs further investigation. Happy to revert it and open a ticket in JIRA if that's better?